### ISSUE TITLE
Add pytest and pytest-benchmark packages to PYPI dependencies list

### ISSUE THREAD

**Issue by chrismgowen** [opencobra/cobrapy#325]
It looks like the new pytest features require pytest and pytest-benchmark for cobra.test.test_all() to work properly. The latter is not included by default at least in my distribution (winpython-3.4.3.7, fwiw). 

Thanks for the improvements!

**Comment by cdiener** [comment]
Thanks for the report. Are you experiencing any issues using cobrapy without pytest installed? Apart from not bein able to run the tests of course :) You are definitely right that the installation instructions include running the tests in `Testing your installation` which is not possible without pytest so that is something we need to fix.

**Comment by chrismgowen** [comment]
No, no other issues beyond that one!

**Comment by pstjohn** [comment]
So, we could add pytest-benchmark to `develop-requirements.txt`, it already contains pytest.

I remember I had a similar issue with this, in that the cobra.test module requires pytest to be installed or throws an `ImportError`. 

Maybe we throw `import pytest` into the test_all function with an appropriate error?
```python
def test_all():
    try:
        import pytest
    except ImportError:
        raise ImportError("cobra test methods require pytest. Run `pip install pytest` "
            "for testing support")
    """ alias for running all unit-tests on installed cobra
    """
    ...
```

**Comment by chrismgowen** [comment]
I installed and upgraded cobra using pip, which can see that it requires `six` based on line 171 in setup.py and various other packages according to the extras dict in that file. You can see these all spelled out in PyPI [here](https://pypi.python.org/pypi/cobra/0.5.6). I don't really have experience with packaging code, so I wouldn't feel comfortable editing any of that, but if it's added to those lists is it correct that those requirements would be taken care of for pip users at least? Maybe that would be an easier and cleaner approach than supporting multiple testing packages?

**Comment by pstjohn** [comment]
Is the use case here that you want to do something like
```
$ pip install cobra
$ python -c 'import cobra.test; cobra.test.test_all()
```
to check the installation?

One reason we'd prefer to not add pytest as a dependency is that we'd like to keep the installation time for cobrapy as small as possible, which typically means reducing the number of dependencies. You don't actually need pytest to run any of the algorithms, so the thought is its only really useful if you've cloned the git repo and are planning on developing cobrapy.

**Comment by phantomas1234** [comment]
Why not just make it a soft dependency and enable something like `pip install cobra[test]` and `pip install cobra[all]`?

**Comment by chrismgowen** [comment]
@pstjohn That's all I was using it for, yes. I don't use it frequently, but I was helping someone else get set up and I thought it would be a convenient way to make sure everything was in place.

**Comment by hredestig** [comment]
fixed in #326 including support for `pip install cobra[test]`
